### PR TITLE
@alloy => another attempt at passing through a request id to gravity

### PR DESCRIPTION
**Pull request by sweir27**
Paired with @orta on this iteration!

It's basically the same concept as the [last PR](https://github.com/artsy/metaphysics/pull/730), but I understand it a bit better and it has tests that execute the entire data-loader-creation flow, which is 👍 .

### CODE REVIEW

File: lib/loaders/loaders_without_authentication/gravity.js [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
-  partnerLoader: gravityLoader(id => `partner/${id}`),
-  partnerShowImagesLoader: gravityLoader(id => `partner_show/${id}/images`),
-  relatedArtworksLoader: gravityLoader("related/artworks"),
-  relatedContemporaryArtistsLoader: gravityLoader("related/layer/contemporary/artists"),
-  relatedFairsLoader: gravityLoader("related/fairs"),
-  relatedMainArtistsLoader: gravityLoader("related/layer/main/artists"),
-  relatedSalesLoader: gravityLoader("related/sales"),
-  relatedShowsLoader: gravityLoader("related/shows"),
-  saleLoader: gravityLoader(id => `sale/${id}`),
-  salesLoader: gravityLoader("sales"),
```
Comment (alloy) [inline lib/loaders/loaders_without_authentication/gravity.js]: @orta It looks like these were not being created on each HTTP request previously?

File: lib/loaders/loaders_without_authentication/positron.js [hunk 2]
```diff
@@ -1,6 +1,13 @@
-import { positronLoaderWithoutAuthenticationFactory as positronLoader } from "../api"
+// @ts-check
 
-export default {
-  articlesLoader: positronLoader("articles"),
-  articleLoader: positronLoader(id => `articles/${id}`),
```
Comment (alloy) [inline lib/loaders/loaders_without_authentication/positron.js]: @orta Same here.

File: lib/apis/gravity.js [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
 import fetch from "./fetch"
+// @ts-ignore
 import config from "config"
 
 const { GRAVITY_API_BASE } = process.env
 
 export default (path, accessToken, fetchOptions = {}) => {
-  const headers = { "X-XAPP-TOKEN": config.GRAVITY_XAPP_TOKEN }
+  const requestID = fetchOptions.requestID
+  const fetchParams = omit(fetchOptions, "requestID")
```
Comment (alloy) [inline lib/apis/gravity.js]: Nit: this can also be `const { requestID, ...fetchParams } = fetchOptions`

**Comment by alloy** [comment]
Looks good, let”s give it a go in staging :+1:

File: test/lib/loaders/request_id.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import { runAuthenticatedQuery, runQuery } from "test/utils"
+import createLoaders from "../../../lib/loaders"
```
Comment (damassi) [inline test/lib/loaders/request_id.js]: We can use an absolute paths -- `lib/loaders`

File: test/lib/loaders/request_id.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import gql from "test/gql"
+
+import factories from "../../../lib/loaders/api"
```
Comment (damassi) [inline test/lib/loaders/request_id.js]: Same here

File: lib/loaders/loaders_without_authentication/positron.js (hunk 2 above)
Comment (orta) [inline lib/loaders/loaders_without_authentication/positron.js]: yeah, looks like the gemini loaders above probably won't be either

File: lib/loaders/loaders_without_authentication/gravity.js (hunk 1 above)
Comment (orta) [reply lib/loaders/loaders_without_authentication/gravity.js]: Can we make a test that `require`s everything under loaders and verifies that it returns a function as default?

Comment (orta) [reply lib/loaders/loaders_without_authentication/gravity.js]: I wrote a PR for it #773

**Pull request merged**